README.md

```diff
@@ -4,30 +4,312 @@
+        'initialize': False,  # Set to True to sync all existing issues
+        'admins': ['admin_username'],  # Email notifications
+    }
+}
+```
+
+### Configuration Options
+
+| Option | Description | Default |
+|--------|-------------|---------|
+| `github_token` | GitHub API token for authentication | Required |
+| `jira` | JIRA instance configurations | Required |
+| `map` | Repository to project mappings | Required |
+| `testing` | Enable dry-run mode (no actual changes) | `False` |
+| `initialize` | Sync all existing issues on startup | `False` |
+| `filters` | Filter issues by labels, milestones, etc. | `{}` |
+| `admins` | List of admin users for notifications | `[]` |
+
+## Usage
+
+### Running the Service
+
+```bash
+# Start the sync service
+sync2jira
+
+# With initialization (sync all existing issues)
+INITIALIZE=1 sync2jira
+
+# Testing mode (dry run)
+# Set testing: True in config file
+sync2jira
+```
+
+### Manual Sync Interface
+
+Access the sync page at `http://your-sync-page-url` to manually trigger synchronization for specific repositories.
+
+### Environment Variables
+
+- `INITIALIZE`: Set to `1` to perform initial sync of all issues
+- `FEDORA_MESSAGING_QUEUE`: Custom message queue name
+- `GITHUB_TOKEN`: GitHub API token (can override config)
+- `JIRA_TOKEN`: JIRA API token (can override config)
+
+## Development
+
+### Setting up Development Environment
+
+```bash
+# Clone and install in development mode
+git clone https://github.com/release-engineering/Sync2Jira.git
+cd Sync2Jira
+pip install -e .
+
+# Install development dependencies
+pip install -r test-requirements.txt
+```
+
+### Running Tests
+
+```bash
+# Run all tests
+tox
+
+# Run specific test environments
+tox -e py39          # Python 3.9 tests
+tox -e lint          # Linting
+tox -e black         # Code formatting check
+tox -e isort         # Import sorting check
+
+# Run tests with coverage
+coverage run -m pytest
+coverage report
+```
+
+### Code Quality
+
+This project uses several tools to maintain code quality:
+
+- **Flake8**: Linting and style checking
+- **Black**: Code formatting
+- **isort**: Import sorting
+- **pytest**: Testing framework
+- **Coverage**: Test coverage reporting
+
+## Documentation
+
+Comprehensive documentation is available at [sync2jira.readthedocs.io](https://sync2jira.readthedocs.io/en/latest/):
+
+- [Quick Start Guide](https://sync2jira.readthedocs.io/en/master/quickstart.html)
+- [Configuration Reference](https://sync2jira.readthedocs.io/en/master/config-file.html)
+- [Adding New Repositories](https://sync2jira.readthedocs.io/en/master/adding-new-repo-guide.html)
+- [Sync Page Usage](https://sync2jira.readthedocs.io/en/master/sync_page.html)
+
+## Contributing
+
+We welcome contributions! Please see our [CONTRIBUTING.md](CONTRIBUTING.md) for details on:
+
+- Setting up your development environment
+- Running tests
+- Submitting pull requests
+- Coding standards
+
+## Common Use Cases
+
```

Comment:
Do we really want this section in here?  (Shouldn't this be in the general documentation?  And, if it's already there, we should _not_ be duplicating it here....)